core/src/main/java/com/linecorp/armeria/server/annotation/BeanRequestConverterFunction.java

```diff
@@ -0,0 +1,546 @@
+ * under the License.
+ */
+package com.linecorp.armeria.server.annotation;
+
+import static com.linecorp.armeria.internal.AnnotatedHttpServiceParamUtil.convertParameter;
+import static com.linecorp.armeria.internal.AnnotatedHttpServiceParamUtil.httpParametersOf;
+import static com.linecorp.armeria.internal.AnnotatedHttpServiceParamUtil.validateAndNormalizeSupportedType;
+import static org.reflections.ReflectionUtils.getAllFields;
+import static org.reflections.ReflectionUtils.getAllMethods;
+
+import java.lang.annotation.Annotation;
+import java.lang.reflect.AnnotatedElement;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Executable;
+import java.lang.reflect.Field;
+import java.lang.reflect.Member;
+import java.lang.reflect.Method;
+import java.lang.reflect.Parameter;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+import java.util.concurrent.ConcurrentMap;
+import java.util.stream.Collectors;
+
+import org.reflections.ReflectionUtils;
+
+import com.google.common.collect.MapMaker;
+
+import com.linecorp.armeria.common.AggregatedHttpMessage;
+import com.linecorp.armeria.common.HttpParameters;
+import com.linecorp.armeria.internal.AnnotatedHttpServiceParamUtil.EnumConverter;
+import com.linecorp.armeria.server.ServiceRequestContext;
+
+import io.netty.util.AsciiString;
+
+/**
+ * A default {@link RequestConverterFunction} which converts a request into a Java bean.
+ *
+ * <p>The following parameters are eligible for automatic injection into bean fields or properties:
+ * <ul>
+ *   <li>Path parameters to fields annotated by {@link Param}</li>
+ *   <li>HTTP parameters to fields annotated by {@link Param}</li>
+ *   <li>Headers to fields annotated by {@link Header}</li>
+ * </ul>
+ */
+public class BeanRequestConverterFunction implements RequestConverterFunction {
+    private static final String PACKAGE_NAME_PREFIX_JAVA = "java.";
+
+    private static final ConcurrentMap<Class<?>, BeanParamInfo> beanParamInfoCache = new MapMaker().weakKeys()
+                                                                                                   .makeMap();
+
+    private static final ConcurrentMap<Class<?>, Boolean> ignoredResultTypeCache = new MapMaker().weakKeys()
+                                                                                                 .makeMap();
+
+    /**
+     * Builds and registers the parameter information for {@code expectedResultType}.
+     *
+     * @param expectedResultType a candidate data type, which will be converted
+     *      by ${@link BeanRequestConverterFunction}
+     */
+    public static synchronized void register(final Class<?> expectedResultType) {
+        final Package pkg = expectedResultType.getPackage();
+        if (pkg != null && pkg.getName().startsWith(PACKAGE_NAME_PREFIX_JAVA)) {
+            // don't handle classes under "java.*" packages
```

Comment:
because:
1. `@RequestObject` could be used on `java.lang.String`, which should be handled by `StringRequestConverterFunction`
2. it also be used on `java.util.Optional<XXX>`, in unit test: https://github.com/line/armeria/blob/2793fc14ddf9b6ba9a6afef07acbe6b9fecfc61e/core/src/test/java/com/linecorp/armeria/server/AnnotatedHttpServiceRequestConverterTest.java#L83-L84
which should be handled by customized request converter `TestRequestConverterOptional1 `
3. on the other hand, in general, our request java bean should never under the "java.*" packages

So, by skip classes under the "java.*" packages, we can make our server start a little faster.

But no big deal, we can totally remove this skip logic, because there are not too many `@RequestObject` annotated classes.

Reply:
removed

Reply:
Thanks for explanation! :heart: